hs_core/views/resource_metadata_rest_api.py

```diff
@@ -0,0 +1,217 @@
+    publisher = serializers.CharField(required=False)
+    sources = SourceSerializer(required=False, many=True)
+    subjects = SubjectSerializer(required=False, many=True)
+    relations = RelationSerializer(required=False, many=True)
+
+    class Meta:
+        model = CoreMetaData
+
+
+class MetadataElementsRetrieveUpdate(generics.RetrieveUpdateDestroyAPIView):
+    """
+    Retrieve resource system (Dublin Core) metadata
+
+    REST URL: /hsapi/resource/{pk}/scimeta/elements/
+    HTTP method: GET
+
+    :type pk: str
+    :param pk: id of the resource
+    :return: resource system metadata as JSON document
+    :rtype: str
+    :raises:
+    NotFound: return json format: {'detail': 'No resource was found for resource id:pk'}
+    PermissionDenied: return json format: {'detail': 'You do not have permission to perform
+    this action.'}
+
+    REST URL: /hsapi/resource/{pk}/scimeta/elements/
+    HTTP method: PUT
+
+    :type pk: str
+    :param pk: id of the resource
+    :type request: JSON formatted string
+    :param request: resource metadata
+    :return: updated resource system metadata as JSON document
+    :rtype: str
+    :raises:
+    NotFound: return json format: {'detail': 'No resource was found for resource id':pk}
+    PermissionDenied: return json format: {'detail': 'You do not have permission to perform
+    this action.'}
+    ValidationError: return json format: {parameter-1': ['error message-1'],
+    'parameter-2': ['error message-2'], .. }
+    """
+    ACCEPT_FORMATS = ('application/json',)
+
+    allowed_methods = ('GET', 'PUT')
+
+    serializer_class = CoreMetaDataSerializer
+
+    def get(self, request, pk):
+        view_utils.authorize(request, pk, needed_permission=ACTION_TO_AUTHORIZE.VIEW_METADATA)
+        resource = hydroshare.get_resource_by_shortkey(shortkey=pk)
+        serializer = CoreMetaDataSerializer(resource.content_object)
+        return Response(data=serializer.data, status=status.HTTP_200_OK)
+
+    def put(self, request, pk):
+        # Update science metadata based on resourcemetadata.xml uploaded
+        resource, authorized, user = view_utils.authorize(
+            request, pk,
+            needed_permission=ACTION_TO_AUTHORIZE.EDIT_RESOURCE,
+            raises_exception=False)
+        if not authorized:
+            raise PermissionDenied()
+
+        metadata = []
+        put_data = request.data
+        keys_to_update = put_data.keys()
+
+        try:
+            if 'title' in keys_to_update:
+                metadata.append({"title": {"value": put_data.pop('title')}})
+
+            if 'description' in keys_to_update:
+                metadata.append({"description": {"abstract": put_data.pop('description')}})
+
+            if 'subjects' in keys_to_update:
+                for subject in put_data.pop('subjects'):
+                    metadata.append({"subject": {"value": subject['value']}})
+
+            hydroshare.update_system_metadata(pk=pk, metadata=metadata)
```

Comment:
Though ```update_system_metadata()``` and ```update_science_metadata()``` functions do serve the same functionality as implemented, I think the implementation of ```update_system_metadata()``` is not correct. The reason being the system metadata is different from science metadata. So it would be better to use ```update_science_metadata()``` here.